docs/_writing_multiple_outputs.rst

```diff
@@ -0,0 +1,248 @@
+
+The `fastqc wrapper <https://github.com/galaxyproject/tools-devteam/blob/master/tools/fastqc/rgFastQC.xml>`__ is an existing tool example where the Java application generates HTML and image outputs but these need to be massaged to make them Galaxy friendly. In other cases, the application or your wrapper must take care of all the fiddly detailed work of writing valid html to display to the user. In either situation, the ``html`` datatype offers a flexible way to display very complex collections of related outputs inside a single history item or to present a complex html page generated by an application. There are some things you need to take care of for this to work:
+
+The following example demonstrates declaring an output of type ``html``.
+
+.. code-block:: xml
+
+    <outputs>
+        <data format="html" name="html_file" label="myToolOutput_${tool_name}.html">
+    </outputs>
+
+The application or script must be set up to write all the output files and/or images to a new special subdirectory passed as a command line parameter from Galaxy every time the tool is run. The paths for images and other files will end up looking something like ``$GALAXY_ROOT/database/files/000/dataset_56/img1.jpg`` when you prepend the Galaxy provided path to the filenames you want to use. The command line must pass that path to your script and it is specified using the ``extra_files_path`` property of the HTML file output.
+
+For example:
+
+
+.. code-block:: xml
+
+
+    <command>myscript.pl "$input1" "$html_file" "$html_file.extra_files_path" </command>
+
+
+The application must create and write valid html to setup the page ``$html_file`` seen by the user when they view (eye icon) the file. It must create and write that new file at the path passed by Galaxy as the ``$html_file`` command line parameter. All application outputs that will be included as links in that html code should be placed in the specific output directory ``$html_file.extra_files_path`` passed on the command line. The external application is responsible for creating that directory before writing images and files into it. When generating the html, The files written by the application to ``$html_file.extra_files_path`` are referenced in links directly by their name, without any other path decoration - eg:
+
+
+.. code-block:: xml
+
+    <a href="file1.xls">Some special output</a>
+    <br/>
+    <img src="image1.jpg" >
+
+The (now unmaintained) Galaxy Tool Factory includes code to gather all output files and create a page with links and clickable PDF thumbnail images which may be useful as a starting point - eg see https://bitbucket.org/fubar/rgalaxy/src/9932187787e592238c2c6fb514a39ff3a705a9af/tools/rgenetics/rgToolFactory.py?at=default
```

Comment:
Perhaps link to here: https://github.com/galaxyproject/tools-iuc/blob/master/tools/tool_factory_2/rgToolFactory2.py#L741 ?